### DISCUSSION TITLE
As a FastAPI's user, I desired error message is more clearly, right now it's only tell me which endpoint respond incorrect model. So, I can fix error at the direct point immediately.

### DISCUSSION THREAD

**Discussion by sila-strike** [fastapi/fastapi#11395] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
async def get_single_task_item(
    # this is wrong using dependency, it should be `Annotated[AsyncSession, Depends(get_async_db)]`
    db: Annotated[AsyncSession, get_async_db],
    id: Annotated[int, Form(alias="task_item_id")],
) -> DBTaskItem:
    pass

@router.post(
    "/",
    response_model=TaskItemAttachment,
    status_code=status.HTTP_201_CREATED,
    dependencies=[Depends(task_item_dependency.is_admin_or_task_assignee)],
)
async def create_task_item_attachment(
    file: Annotated[UploadFile, File()],
    service: TaskItemAttachmentServiceDependency,
    task_item: Annotated[
        DBTaskItem,
        Depends(get_single_task_item),
    ],
):
    pass
```

### Description

![image](https://github.com/tiangolo/fastapi/assets/162951068/c0b40872-fbfc-4c65-8cb5-1eaa0cc46090)

I desired the error to tell me that "the `db: Annotated[AsyncSession, get_async_db]` is not correct type".

Is it possible to display the error message more clearly?

**Related libraries:**
FastAPI v.0.103.*
Pydantic v.1.8.*

### Operating System

macOS

### FastAPI Version

0.103.2

### Pydantic Version

1.8.2

### Python Version

3.10.15

### Additional Context

In the example code above, I used wrong `db` parameter statement

It should be `Annotated[AsyncSession, Depends(get_async_db)]` instead of `db: Annotated[AsyncSession, get_async_db]`.  
I need the error message to point to this line rather than just point to the endpoint.

**Comment by YuriiMotov** [comment]
You can simplify this example as follows
```
from typing import Annotated
from fastapi import FastAPI
from sqlalchemy.ext.asyncio import AsyncSession

app = FastAPI()

def get_async_db():
    pass

@app.post("/")
def get_single_task_item(
    db: Annotated[AsyncSession, get_async_db],
) -> str:
    pass
```

or without SqlAlchemy:
```
from typing import Annotated
from fastapi import FastAPI

app = FastAPI()

class A:
    def __init__(*args, **kwargs):
        pass

@app.post("/")
def get_single_task_item(
    a: Annotated[A, ""],
) -> str:
    pass
```

And the error message is really confusing (fastapi 0.110.1, pydantic 2.6.1):

> fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that typing.Annotated[t.A, ''] is a valid Pydantic field type. If you are using a return type annotation that is not a valid Pydantic field (e.g. Union[Response, dict, None]) you can disable generating the response model from the type annotation with the path operation decorator parameter response_model=None. Read more: https://fastapi.tiangolo.com/tutorial/response-model/

Looks like FastAPI treats that input parameter as a response.
